_includes/tutorials/message-ordering/kafka/code/src/main/java/io/confluent/developer/KafkaProducerApplication.java

```diff
@@ -0,0 +1,111 @@
+        return producer.send(producerRecord);
+    }
+
+    public void shutdown() {
+        producer.close();
+    }
+
+    public static Properties loadProperties(String fileName) throws IOException {
+        final Properties envProps = new Properties();
+        final FileInputStream input = new FileInputStream(fileName);
+        envProps.load(input);
+        input.close();
+
+        return envProps;
+    }
+
+    public void printMetadata(final Collection<Future<RecordMetadata>> metadata,
+                              final String fileName) {
+        System.out.println("Offsets and timestamps committed in batch from " + fileName);
+        metadata.forEach(m -> {
+            try {
+                final RecordMetadata recordMetadata = m.get();
+                System.out.println("Record written to offset " + recordMetadata.offset() + " timestamp " + recordMetadata.timestamp() + " partition " + recordMetadata.partition());
+            } catch (InterruptedException | ExecutionException e) {
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+        });
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length < 2) {
+            throw new IllegalArgumentException(
+                    "This program takes two arguments: the path to an environment configuration file and" +
+                            "the path to the file with records to send");
+        }
+
+        final Properties props = KafkaProducerApplication.loadProperties(args[0]);
+
+        props.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, "true");
+        props.put(ProducerConfig.ACKS_CONFIG, "all");
+
+        props.put(ProducerConfig.CLIENT_ID_CONFIG, "myApp");
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+        final String topic = props.getProperty("output.topic.name");
+        final Producer<String, String> producer = new KafkaProducer<>(props);
+        final KafkaProducerApplication producerApp = new KafkaProducerApplication(producer, topic);
+
+        // Attach shutdown handler to catch Control-C.
+        Runtime.getRuntime().addShutdownHook(new Thread(producerApp::shutdown));
+        String filePath = args[1];
+        try {
+            List<String> linesToProduce = Files.readAllLines(Paths.get(filePath));
+            List<Future<RecordMetadata>> metadata = linesToProduce.stream()
+                    .filter(l -> !l.trim().isEmpty())
+                    .map(producerApp::produce)
+                    .collect(Collectors.toList());
+            producerApp.printMetadata(metadata, filePath);
+
+        } catch (IOException e) {
+            System.err.println(String.format("Error reading file %s due to %s", filePath, e));
+        }
+        producerApp.shutdown();
```

Comment:
Since a shutdown hook is configured, this additional call to `shutdown` isn't needed.

Reply:
@mikebin , ok, I can validate and remove the call to `shutdown()`.

@bbejeck similar to the previous comment, this code is also present in other KTs. Assuming you agree with this removal, do you have a preference whether this PR addresses those KTs or should I file a separate GH issue?

Reply:
@mikebin  good catch

>@bbejeck similar to the previous comment, this code is also present in other KTs. Assuming you agree with this removal, do you have a preference whether this PR addresses those KTs or should I file a separate GH issue?

that was an oversight on my part during refactoring, I'll clean it up in a separate PR

Reply:
Note to self: mirror changes that @bbejeck did in https://github.com/confluentinc/kafka-tutorials/pull/594

Reply:
Addressed by https://github.com/confluentinc/kafka-tutorials/pull/587/commits/44f2b420df1bc802781c0dc7e9c8fde322691f3b